fido/fido.py

```diff
@@ -110,21 +152,45 @@ def finished_errorback(error):
 
     # Fetch the body once we've received the headers
     def response_callback(response):
+        finished = Deferred()
         response.deliverBody(HTTPBodyFetcher(response, finished))
+        return finished
 
     deferred.addCallback(response_callback)
-    deferred.addErrback(finished.errback)
 
-    if timeout is not None:
-        # Cancel the request if we hit the timeout
-        def cancel_timer(response):
-            if timer.active():
-                timer.cancel()
-            return response
-        timer = reactor.callLater(timeout, deferred.cancel)
-        finished.addBoth(cancel_timer)
+    # erroback which handles various types of twisted timeout errors
```

Comment:
You should probably add a docstring that state that we could let these errors bubble up, but for sanity of usage, we are raising more user friendly crochet.TimeoutError exception to hide the twisted ones.